### PR TITLE
Completed sgt-back-end challenge -Kaitlin Lovrich

### PR DESCRIPTION
**Pull request by kaitlin-lovrich**

### CODE REVIEW

File: sgt-back-end/server.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    const { name, course, score } = req.body;
+    const sql = `
+      insert into "grades" ("name", "course", "score")
+        values ($1, $2, $3)
+        returning *;
+    `;
+    const params = [name, course, score];
+    const result = await db.query(sql, params);
+    const grade = result.rows;
+    if (!name || !course || !score) {
```
Comment (RobertGardner) [inline sgt-back-end/server.ts]: You should do this check _before_ issuing the SQL. If the input data is bad, you don't want to put it in the database -- and the database may throw an exception which will result in a 500 error, so the client will never learn that they sent you bad data.

File: sgt-back-end/server.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+});
+
+app.put('/api/grades/:gradeId', async (req, res, next) => {
+  try {
+    const { name, course, score } = req.body;
+    const gradeId = Number(req.params.gradeId);
+    console.log(gradeId);
+    if (!Number.isInteger(gradeId) || gradeId <= 0) {
+      throw new ClientError(400, '"gradeId" must be a positive integer');
+    }
```
Comment (RobertGardner) [inline sgt-back-end/server.ts:114]: As in the post, you should validate that the name, course, score are valid.

**Review by RobertGardner — APPROVED** [review]
I'm impressed! Not only were you first, but your code is excellent!

**Comment by kaitlin-lovrich** [comment]
Thank you 🥹😁

**Pull request merged**